contracts/verifiers/UniversalVerifierMultiQuery.sol

```diff
@@ -0,0 +1,927 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.27;
+
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {Ownable2StepUpgradeable} from "@openzeppelin/contracts-upgradeable/access/Ownable2StepUpgradeable.sol";
+import {IRequestValidator} from "../interfaces/IRequestValidator.sol";
+import {IAuthValidator} from "../interfaces/IAuthValidator.sol";
+import {IState} from "../interfaces/IState.sol";
+
+contract UniversalVerifierMultiQuery is Ownable2StepUpgradeable {
+    /**
+     * @dev Version of the contract
+     */
+    string public constant VERSION = "1.0.0";
+
+    /**
+     * @dev Request. Structure for request for storage.
+     * @param metadata Metadata of the request.
+     * @param validator Validator circuit.
+     * @param params Params of the request. Proof parameters could be ZK groth16, plonk, ESDSA, EIP712, etc.
+     */
+    struct RequestData {
+        string metadata;
+        IRequestValidator validator;
+        bytes params;
+    }
+
+    struct Request {
+        uint256 requestId;
+        string metadata;
+        IRequestValidator validator;
+        bytes params;
+    }
+
+    struct GroupedRequests {
+        uint256 groupId;
+        Request[] requests;
+    }
+
+    /**
+     * @dev Struct to store proof and associated data
+     */
+    struct Proof {
```

Comment:
One more comment here. It doesn't look correct to use the same Proof structure for both auth and request proofs.
They serve different purpose.
I suggest use different structure for those two and at least `metadata` mapping is not there for auth proof information